### ISSUE TITLE
sidekiqctl:77:in `kill': No such process (Errno::ESRCH)

### ISSUE THREAD

**Issue by larryzhao** [mperham/sidekiq#2766]
Hi, Guys, 

Sorry for the bothering.

I encounter this error from time to time during deploying. 

At the end of deploying I will issue `~/.rvm/bin/rvm 2.3.0@mygemset do bundle exec sidekiqctl stop /home/deployer/deploy/myapp/shared/pids/sidekiq.pid` on the server, and not every time, but some times I will get the following exception on one of the servers:

```
/home/deployer/deploy/myapp/shared/bundle/ruby/2.3.0/gems/sidekiq-3.5.3/bin/sidekiqctl:77:in `kill': No such process (Errno::ESRCH)
  from /home/deployer/deploy/myapp/shared/bundle/ruby/2.3.0/gems/sidekiq-3.5.3/bin/sidekiqctl:77:in `stop'
  from /home/deployer/deploy/myapp/shared/bundle/ruby/2.3.0/gems/sidekiq-3.5.3/bin/sidekiqctl:37:in `initialize'
  from /home/deployer/deploy/myapp/shared/bundle/ruby/2.3.0/gems/sidekiq-3.5.3/bin/sidekiqctl:92:in `new'
  from /home/deployer/deploy/myapp/shared/bundle/ruby/2.3.0/gems/sidekiq-3.5.3/bin/sidekiqctl:92:in `<top (required)>'
  from /home/deployer/deploy/myapp/shared/bundle/ruby/2.3.0/bin/sidekiqctl:23:in `load'
  from /home/deployer/deploy/myapp/shared/bundle/ruby/2.3.0/bin/sidekiqctl:23:in `<main>'
```

I've also tried to check on the server before issuing `sidekiqctl stop`. Pid file path, the process id in pid file were both correct, but the error still occured.
Why is that?

Thanks for helping.

Rails: 4.2.5
Sidekiq: 3.5.3
Ruby: MRI 2.3.0
Server: Ubuntu 12.04

**Comment by mperham** [comment]
That doesn't make any sense to me.  Line 77 doesn't call Process.kill and I rescue ESRCH everywhere I do.

**Comment by larryzhao** [comment]
@mperham Ah, it's `Process.kill(:KILL, pid)` in version 3.5.3. In master branch of `3.x`, it's now using `kill -9`. I think it will be fixed next version. Am I right?

**Comment by mperham** [comment]
Thank you for tracking that down and reminding me.  As a reward I have released 3.5.4 with this fix in it.  :+1:

**Comment by larryzhao** [comment]
Great news. Thx!
